### DISCUSSION TITLE
Is `setlocal` not honored by VSCode NeoVim?

### DISCUSSION THREAD

**Discussion by lindhe** [vscode-neovim/vscode-neovim#2322] [Q&A]
I'm not sure if this is a bug or just me misunderstanding (Neo)Vim, so I thought I'd ask here before creating an issue…

So I noticed that VSCode NeoVim seemingly randomly started setting `spell` in my editors and I couldn't figure out why. It's turned off by default for me. Turns out I had `setlocal spell` in my `after/ftplugin` config for markdown files, so every time I opened a markdown file VSCode NeoVim would activate `spell` for me. Mystery solved.

The thing is: I used `setlocal` because I was under the impression that this would ensure that only the buffer where the markdown file was open would get `spell` activated, instead of all buffers. Is this not the case?? If it is the case, then I think it seems like VSCode NeoVim does not honor the `setlocal` properly.

Any input on this would be appreciated!

Cheers!

**Comment by justinmk** [comment]
> Turns out I had `setlocal spell` in my `after/ftplugin` config for markdown files, so every time I opened a markdown file VSCode NeoVim would activate `spell` for me. Mystery solved.
> 
> The thing is: I used `setlocal` because I was under the impression that this would ensure that only the buffer where the markdown file was open would get `spell` activated

Possibly it's getting inherited somehow. Note that vscode-neovim uses buftype=nofile buffers, maybe there is a detail that needs to be considered.

**Reply by lindhe** [reply]
I'm utterly ignorant of `buftype`, but it sounds like a good guess. Is there anything I can do to try and work around it, or do you consider this a bug that should be fixed in vscode-neovim?

If you think it's a bug: should I create an Issue for it?
